### PR TITLE
Added code to check the appropriate functioning of ft_spectralanalysis

### PR DESCRIPTION
**Pull request by hadrianswall**
Plus many more small changes (most of them really obvious; life is not only about documenting but also about doing(-)

Added a test script.

### CODE REVIEW

File: contrib/misc/ft_laggedcoherence.m [hunk 1]
```diff
@@ -169,10 +183,18 @@
...
+% requested cfg_freq.foi (calculated as freq) using the requested number of cycles (cfg.numcycles).
+% 
+% cyclelengths=t_ftimwin/cfg.numcycles
+% correspondingfreqs=ones(size(cyclelengths))./cyclelengths
+% freq % compare with correspondingfreqs
+
   freqout{freqindx} = ft_freqanalysis(cfg_freq,datain);
   
-  cfg_lcoh.laggedcoherence.lags = cfg.numcycles.*cfg.loi./freqout{freqindx}.freq;
+fr  cfg_lcoh.laggedcoherence.lags = cfg.numcycles.*cfg.loi./freqout{freqindx}.freq;
```
Comment (schoffelen) [inline contrib/misc/ft_laggedcoherence.m]: I guess that the 'fr' is a typo?

File: contrib/misc/testlaggedcoherence.m [hunk 2]
```diff
@@ -0,0 +1,12 @@
+datain=[];
```
Comment (schoffelen) [inline contrib/misc/testlaggedcoherence.m]: This function is a duplicate of the function in fieldtrip/test, called test_laggedcoherence.m Perhaps best to make the changes there, and remove this file from the contrib/misc/ folder

File: contrib/misc/ft_laggedcoherence.m (hunk 1 above)
Comment (hadrianswall) [reply contrib/misc/ft_laggedcoherence.m]: Met dikke vingers getikt, maar het punt dat ik wilde maken betreft de code die daarvoor staat ...

File: contrib/misc/testlaggedcoherence.m (hunk 2 above)
Comment (hadrianswall) [reply contrib/misc/testlaggedcoherence.m]: OK. Hoe had ik dit kunnen weten?

Comment (schoffelen) [reply contrib/misc/testlaggedcoherence.m]: niet. 

But it seems that the equivalent fieldtrip/test/test_laggedcoherence looks uncannily similar to the testlaggedcoherence in contrib/misc. So, perhaps I already moved it over at some point.

**Comment by schoffelen** [comment]
I'll merge this for now, and then will move the testlaggedcoherence to fieldtrip/test.

**Comment by schoffelen** [comment]
Note that there is still the open issue related to a mismatch between expected frequency bins (by Eric :) ), and the actual outputted frequency bins

**Pull request merged**